hieroplatform/src/main/java/org/hiero/sketch/spreadsheet/BasicColStat.java

```diff
@@ -0,0 +1,121 @@
+package org.hiero.sketch.spreadsheet;
+
+import org.hiero.sketch.table.api.IColumn;
+import org.hiero.sketch.table.api.IMembershipSet;
+import org.hiero.sketch.table.api.IRowIterator;
+import org.hiero.sketch.table.api.IStringConverter;
+import javax.annotation.Nullable;
+
+public class BasicColStat {
+    private final int MOMENT_NUM;
+    private double min;
+    @Nullable private Object minObject;
+    private double max;
+    @Nullable private Object maxObject;
+    private final double moments[];
+    private long size;
+
+    public BasicColStat(int momentNum) {
+        if (momentNum < 1)
+            throw new IllegalArgumentException("number of moments cannot be positive");
+        this.MOMENT_NUM = momentNum;
+        this.moments = new double[this.MOMENT_NUM];
+        this.min = Double.POSITIVE_INFINITY;
+        this.max = Double.NEGATIVE_INFINITY;
+    }
+
+    public BasicColStat() {
+        this.MOMENT_NUM = 2;
+        this.moments = new double[this.MOMENT_NUM];
+        this.min = Double.POSITIVE_INFINITY;
+        this.max = Double.NEGATIVE_INFINITY;
+    }
+
+    public double getMin() { return this.min; }
+
+    public Object getMinObject() {
+        assert this.minObject != null;
+        return this.minObject;
+    }
+
+    public double getMax() { return this.max; }
+
+    public Object getMaxObject() {
+        assert this.maxObject != null;
+        return this.maxObject;
+    }
+
+    /**
+     *
+     * @param i a number in {0,1,...,MOMENT_NUM}
+     * @return the i'th moment which is the sum of x^i
+     */
+    public double getMoment(int i) {
+        if ((i < 1) || (i > this.MOMENT_NUM))
+            throw new IllegalArgumentException("moment requested doesn't exist");
+        return this.moments[i - 1];
+    }
+
+    public long getSize() { return this.size; }
```

Comment:
This deserves a JavaDoc comment. This is the size of what?
Perhaps it should be called something else, e.g. rowCount?